src/uk/co/llobile/kotlin/hours/types/Monad.kt

```diff
@@ -0,0 +1,8 @@
+package uk.co.llobile.kotlin.hours.types
+
+/**
+ * Definition of a Monad.
+ */
+interface Monad<T> {
+    fun <R> lift(fn: (T) -> Monad<R>): Monad<R>
```

Comment:
Historically the function is called _flatMap_. It should also provide a _unity()_

Reply:
Wikipedia suggests flatMap() and join(). I'll probably go with that.